### ISSUE TITLE
mv doesn't expand ~

### ISSUE THREAD

**Issue by nibon7** [nushell/nushell#6151]
### Describe the bug

* mv

```
/data/source/nushell〉touch a.txt                                                                                                        07/27/2022 11:44:26 PM
/data/source/nushell〉mv -v a.txt ~                                                                                                      07/27/2022 11:44:33 PM
╭───┬────────────────────────────────────────────────────────────╮
│ 0 │ moved /data/source/nushell/a.txt to /data/source/nushell/~ │
╰───┴────────────────────────────────────────────────────────────╯
/data/source/nushell〉touch a.txt                                                                                                        07/27/2022 11:44:38 PM
/data/source/nushell〉mv -v a.txt ~/b.txt                                                                                                07/27/2022 11:44:51 PM
Error:
  × Could not move "/data/source/nushell/a.txt" to "/data/source/nushell/~/b.txt". Io error. Look inside err_kind for more details.
   ╭─[entry #5:1:1]
 1 │ mv -v a.txt ~/b.txt
   ·       ──┬──
   ·         ╰── could not move
   ╰────

```

* cp

```
/data/source/nushell〉touch a.txt                                                                                                        07/28/2022 12:16:07 AM
/data/source/nushell〉cp -v a.txt ~                                                                                                      07/28/2022 12:16:24 AM
╭───┬─────────────────────────────────────────────────────────────╮
│ 0 │ copied /data/source/nushell/a.txt to /data/source/nushell/~ │
╰───┴─────────────────────────────────────────────────────────────╯
/data/source/nushell〉cp -v a.txt ~/                                                                                                     07/28/2022 12:16:27 AM
Error: nu::shell::directory_not_found (link)

  × Directory not found
   ╭─[entry #15:1:1]
 1 │ cp -v a.txt ~/
   ·             ─┬
   ·              ╰── directory not found
   ╰────
  help: destination directory does not exist

/data/source/nushell〉cp -v a.txt ~/a.txt                                                                                                07/28/2022 12:16:30 AM
Error: nu::shell::file_not_found (link)

  × File not found
   ╭─[entry #16:1:1]
 1 │ cp -v a.txt ~/a.txt
   · ─┬
   ·  ╰── copy file "/data/source/nushell/a.txt" failed: Not a directory (os error 20)
   ╰────
/data/source/nushell〉cp a.txt ~/a.txt                                                                                                   07/28/2022 12:18:43 AM
/data/source/nushell〉^ls ~/a.txt                                                                                                        07/28/2022 12:18:59 AM
ls: cannot access '/home/nibon7/a.txt': No such file or directory
/data/source/nushell〉^ls *.txt                                                                                                          07/28/2022 12:19:07 AM
a.txt  c.txt  README.release.txt
```

**EDIT**: `cp` doesn't expand `~` too. It seems that `cp` is weird.

### How to reproduce

1. touch a.txt
2. mv -v a.txt ~/a.txt

### Expected behavior

mv should expand ~ to home dir

### Configuration

| key                | val                                      |
| ------------------ | ---------------------------------------- |
| version            | 0.66.1                                   |
| branch             | main                                     |
| commit_hash        | c92211c01689a4eb76c5a3f755506ab4bf9f8804 |
| build_os           | linux-x86_64                             |
| build_target       | x86_64-unknown-linux-gnu                 |
| rust_version       | rustc 1.62.0 (gentoo)                    |
| cargo_version      | cargo 1.62.0                             |
| pkg_version        | 0.66.1                                   |
| build_time         | 2022-07-27 05:53:25 +00:00               |
| build_rust_channel | debug                                    |
| features           | default, trash, which, zip               |
| installed_plugins  |                                          |

**Comment by nibon7** [comment]
Maybe `ls`,`cp` and `mv` should be reviewed again :joy:

**Comment by fdncred** [comment]
ugh, haven't we been around and around with should it expand or shouldn't it expand?

**Comment by nibon7** [comment]
I found just now and it seems the old implementation is ok.

**Comment by fdncred** [comment]
Please look at the old PRs and see what was changed. I'm not up for changing this every release.

**Comment by nibon7** [comment]
Ok. I’ll check it later.

**Comment by nibon7** [comment]
> Please look at the old PRs and see what was changed. I'm not up for changing this every release.

Before commit 7d46177cf373ae256cde23e197108b852c3aadc5, it's ok. By the way, I prefer something like `[source1 source2 source3] | rm`, `[source1 source2 source3] | mv target_dir` and `[source1 source2 source3] | cp target_dir` to handle multiple files.

**Comment by fdncred** [comment]
Does this come into play? https://github.com/nushell/nushell/pull/6014

**Comment by nibon7** [comment]
> Does this come into play? #6014

I said the `mv` is builtin `mv` command and I don't think it has anything to do with this PR. The relevant PR probaly is #6103.

**Comment by nibon7** [comment]
Fixed by #6169
